File: gas-station.rs
fn check(i:usize, gas: &Vec<i32>, cost: &Vec<i32>) -> bool {
    let mut total = gas[i];
    for x in 1usize..= gas.len() {
        let next = (x +i) % gas.len();
        let prev = (next + gas.len() - 1) % gas.len();
        total -= cost[prev];
        if total <  0 {
            return false
        }
        total += gas[next]
    }
    true
}

fn gas_truck(gas: Vec<i32>, cost: Vec<i32>) -> i32 {
    for i in 0..gas.len() {
        if check(i, &gas, &cost) {
            return i as i32
        }
    }
    -1
}


fn main() {
    let a = vec![1,2,3,4,5];
    let b = vec![3,4,5,1,2];
    println!("{}", gas_truck(a, b))
}

File: gas-station.js
const check = (i, gas, cost) => {
    let total = gas[i]
    const len = gas.length
    let ret = true
    for(let x = 1; x < len + 1 ; x++) {
        const next = (x + i) % len
        const prev = (next + len - 1) % len
        total -= cost[prev]
        if (total < 0) {
            ret = false
        }
        total += gas[next]
    }
    return ret
}

const gasStation = (gas, cost) => {
    const len = gas.length
    let index = -1
    for(let i = 0; i < len; i++) {
        if(check(i, gas, cost)) {
            index = i
            break
        }
    }
    return index
}


const gas = [1,2,3,4,5]
const cost = [3,4,5,1,2]
console.log(gasStation(gas, cost))


